File: springboot2-junit5/src/test/java/cc/before30/home/junit5/Junit5ApplicationTest.java
package cc.before30.home.junit5;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.web.client.TestRestTemplate;
import org.springframework.test.context.junit.jupiter.SpringExtension;

import static org.junit.jupiter.api.Assertions.*;

@ExtendWith(SpringExtension.class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class Junit5ApplicationTest {

    @Autowired
    private TestRestTemplate restTemplate;

    @Test
    @DisplayName("test Message REST API")
    void testMessage() {
        String message = this.restTemplate.getForObject("/hello", String.class);
        assertEquals("world", message);
    }

}

File: springboot2-junit5-kt/src/test/kotlin/cc/before30/home/junit5/Junit5ApplicationTest.kt
package cc.before30.home.junit5

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.web.client.TestRestTemplate
import org.springframework.test.context.junit.jupiter.SpringExtension

@ExtendWith(SpringExtension::class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
internal class Junit5ApplicationTest {
    @Autowired
    lateinit var testRestTemplate: TestRestTemplate

    @Test
    @DisplayName("test Message REST API")
    fun `test Message REST API`() {
        val message = testRestTemplate.getForObject("/hello2", String::class.java)
        assertEquals("world2", message)
    }
}